### DISCUSSION TITLE
deploy endpoint in complete local mode

### DISCUSSION THREAD

**Discussion by anotinelg** [aws/sagemaker-python-sdk#2349] [Help]
Is it possible to deploy locally an endpoint without any connection with s3?

With local code, and local instance, i can not get it work. 
I look at the following code, and it seems that even if i have local_code, if the repack value is True, it will try to upload to s3 the repacked artefacts.

https://github.com/aws/sagemaker-python-sdk/blob/c4d71f5639697d833e5578016a3f45402a4a80de/src/sagemaker/model.py#L1117

**Comment by ChoiByungWook** [comment]
Hey @anotinelg,

Thanks for bringing this issue to our attention.

This looks like a gap in our Python SDK for supporting true local mode when using our Framework Estimator/Model.

There doesn't seem to be a way to disable repack_model through the Framework Estimator/Model constructors.

I think one workaround that may work is to use the Estimator/Model class and point the `image_uri` to the corresponding image that is used in the Framework Estimator you were attempting to run local mode with, as the base Model class shouldn't call repack_model.

https://github.com/aws/sagemaker-python-sdk/blob/c4d71f5639697d833e5578016a3f45402a4a80de/src/sagemaker/model.py#L282

I will note this in backlog, which is prioritized based on customer feedback.

**Comment by anotinelg** [comment]
Actually i am using the MxnetModel which will use the Framework Model and hence the method prepare_container_def which calls the upload_code. I am not sure i understand what you propose. Instead of using MXNetModel, i should directly use the Model class?

**Comment by ChoiByungWook** [comment]
> Actually i am using the MxnetModel which will use the Framework Model and hence the method prepare_container_def which calls the upload_code. I am not sure i understand what you propose. Instead of using MXNetModel, i should directly use the Model class?

Yeah, the MXNetModel class ends up extending the [FrameworkModel](https://github.com/aws/sagemaker-python-sdk/blob/c4d71f5639697d833e5578016a3f45402a4a80de/src/sagemaker/model.py#L878) class, which then extends the [Model](https://github.com/aws/sagemaker-python-sdk/blob/c4d71f5639697d833e5578016a3f45402a4a80de/src/sagemaker/model.py#L41) class.

I believe the base Model class doesn't call repack_model, however you would lose out on a few things that MXNetModel class does for you such as the image_uri, repacking of the model with your inference script. These things you would have to supply to the base Model class.
